File: tests/fixtures/sample.ts
import { Request, Response } from 'express';
import { UserService } from './user.service';

export interface UserDTO {
  id: string;
  name: string;
  email: string;
}

export type UserRole = 'admin' | 'user' | 'guest';

export class UserController {
  constructor(private userService: UserService) {}

  async getUser(req: Request, res: Response): Promise<void> {
    const user = await this.userService.findById(req.params.id);
    res.json(user);
  }

  async createUser(req: Request, res: Response): Promise<void> {
    const user = await this.userService.create(req.body);
    res.status(201).json(user);
  }
}

export function validateEmail(email: string): boolean {
  return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
}

const formatUser = (user: UserDTO): string => {
  return `${user.name} <${user.email}>`;
};

export default UserController;

File: tests/fixtures/sample.rs
use std::collections::HashMap;

pub struct User {
    pub id: u64,
    pub name: String,
    pub email: String,
}

pub enum UserRole {
    Admin,
    User,
    Guest,
}

pub trait Repository {
    fn find_by_id(&self, id: u64) -> Option<&User>;
    fn find_all(&self) -> Vec<&User>;
}

impl User {
    pub fn new(id: u64, name: String, email: String) -> Self {
        Self { id, name, email }
    }
}

pub async fn fetch_user(id: u64) -> Result<User, String> {
    Ok(User::new(id, "Test".into(), "test@example.com".into()))
}

fn validate_email(email: &str) -> bool {
    email.contains('@') && email.contains('.')
}
